editor/state.js

```diff
@@ -34,35 +40,49 @@ export const blocks = combineReducers( {
 		switch ( action.type ) {
 			case 'REPLACE_BLOCKS':
 				return action.blockNodes.map( ( { uid } ) => uid );
+			case 'INSERT_BLOCK':
```

Comment:
Maybe not here, but we'll need to consider how to insert a block at a specific index, supporting behavior like in this mockup:

![Insert at index](https://cloud.githubusercontent.com/assets/1779930/24963588/02d6c9ec-1f6d-11e7-99de-f72004032e05.png)

https://wpcoredesign.mystagingwebsite.com/gutenberg/